openedx/openedx-platform
# feat: allow instructors to manage certificates

## Description

Currently only super admin and staff are able to configure the certificates for the course.

This PR allows course instructors (admin course permissions) to manage certificates through all the actions available in the tab instructor > certificates, in this way, each instructor will be able to manage their certificates, which allows more granularity on the permissions granted to users. The staff permission is a somewhat dangerous permission, since it grants access to a wide variety of functions, with this change the instructors (who already manage the users, cohorts, and others) will also be able to manage the certificates. 

This feature is used for eduNEXT due to client requests and we think is useful for the community, we put this into consideration for product reviewers (@santiagosuarezedunext @jmakowski1123 ).

## Testing instructions
1. Enable these `FEATURES` in your environment (to activate all certificate options):

```yml
  CERTIFICATES_INSTRUCTOR_GENERATION: true
  CUSTOM_CERTIFICATE_TEMPLATES_ENABLED: true
```
2. Create a course with a certificate.
3. Create a new user without special permissions (superuser, staff).
4. Add the user to the course team as Admin (i.e. Instructor) via the instructor > membership tab.
![image](https://user-images.githubusercontent.com/66016493/200370081-d05e3319-3e40-4883-8e19-774cd5c57715.png)
5. The certificate tab should be visible and all the certificate operations should work (certificate generation, whitelist, self generation, etc)
![Screenshot from 2022-11-04 20-38-01](https://user-images.githubusercontent.com/66016493/200371064-7f8f7bfd-90fe-4389-b29c-d6d8c80842f2.png)

C.C @MaferMazu @Alec4r @felipemontoya @Ian2012
